src/rust/engine/src/nodes.rs

```diff
@@ -1205,12 +1209,14 @@ impl Node for NodeKey {
 
   fn run(self, context: Context) -> NodeFuture<NodeResult> {
     let span_id = generate_random_64bit_string();
-    let node_workunit_params = if context.session.should_record_zipkin_spans() {
+    let node_workunit_params = {
```

Comment:
This is a very performance sensitive codepath (which is why it used an option before): would suggest moving the `format!`/`start_time`/etc inside a conditional here, as before.

But also: IMO, the "display name" here should replace the name, and nodes should only actually be rendered if they are named.